### PR TITLE
Add gray box to My PL page with links

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Finishes https://codedotorg.atlassian.net/browse/ACQ-1429 and moves links from the teacher homepage to the Professional Landing page.

Admin View:
Screenshot 2024-04-22 at 11 13 15 AM

Workshop Admin view:
Screenshot 2024-04-22 at 11 14 44 AM

### CODE REVIEW

File: dashboard/app/views/home/_homepage.html.haml [hunk 1]
```diff
@@ -23,22 +23,10 @@
 #homepage-container
 
 #workshop-links.container.main
-  -if can? :read, Pd::Application::ApplicationBase
```
Comment (dmcavoy) [inline dashboard/app/views/home/_homepage.html.haml]: These links should to more than Workshop Admins and until the work to add the Links for RPs, Facilitators and Workshop organizers is done I don't think we can remove

Comment (dmcavoy) [reply dashboard/app/views/home/_homepage.html.haml]: Oh I see what you did. You are allows the same people to see the gray box. I don't think we want RPs, Facilitators and Workshop Organizers to see that gray box. Just internal Workshop Admins

Comment (bethanyaconnor) [reply dashboard/app/views/home/_homepage.html.haml]: oh woops -- I misunderstood! I'll move this PR back to draft and update

Comment (bethanyaconnor) [reply dashboard/app/views/home/_homepage.html.haml]: @dmcavoy circling back to this. In [the spec](https://docs.google.com/document/d/1Ee02kw_ei2x6wOS06K768dC-kUhGDlydjoE6BZipbPE/edit#bookmark=kix.o5cbqsv9l6ei), these are the links listed to move to the gray box. I see that the application dashboard, workshop dashboard, etc are moving to other places as well. Should those links not be in the gray box at all?

Comment (dmcavoy) [reply dashboard/app/views/home/_homepage.html.haml]: No issues with the links. I think you have the right list of links. What I was commenting on was the permissions. As a whole this gray box should only show for Workshop Admins (internal Code.org staff who manage workshop and application stuff). Workshop Organizers, Regional Partners, Facilitators, etc will all have access to these links right on the page. Does that help?

Comment (bethanyaconnor) [reply dashboard/app/views/home/_homepage.html.haml]: I think so!

File: dashboard/app/views/pd/professional_learning_landing/_pl_extra_links.haml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+- extra_links = []
+
+
```
Comment (bethanyaconnor) [inline dashboard/app/views/pd/professional_learning_landing/_pl_extra_links.haml:3]: I decided to keep the conditions for which links are shown the same as they are on [the homepage](https://github.com/code-dot-org/code-dot-org/blob/daf0ae6d9ed8e3044a6e8ca89ab0edb278497b76/dashboard/app/views/home/_homepage.html.haml#L25), but limit this box to admins and workshop admins. Likely some redundancy here, so I'm open to push back

**Review by TurnerRiley — APPROVED** [review]
Looks great! I think the redundancy seems fine especially since it's covered by the great tests you added!

**Pull request merged**